pkg/router/controller/controller.go

```diff
@@ -97,7 +114,21 @@ func (c *RouterController) HandleEndpoints() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// Change the local sync state within the lock to ensure that all
+	// event handlers have the same view of sync state.
+	c.endpointsReplaceRead = c.EndpointsReplaceRead()
```

Comment:
I'm assuming you meant 'just not seeing'.  A sync from etcd never results in router state being cleared.  The new stream of events are used to modify the existing router state.